### DISCUSSION TITLE
Upgrade to newer version of ansible-lint?

### DISCUSSION THREAD

**Discussion by Errosion** [ansible/ansible-lint#1508] [Q&A] [answered]
I am running ansible-lint on a CentOS 7 platform and upon following the installation instructions (https://ansible-lint.readthedocs.io/en/latest/installing.html), the version of ansible-lint that got installed is 4.2.0. I am also using the base python version that comes with CentOS 7, 2.7.5. This is my first installation/instance of ansible-lint in our environment, so I apologize if this comes off a bit simplistic.

If I attempt to do a `pip install upgrade "ansible-lint"` I receive these messages:
     ERROR: Could not find a version that satisfies the requirement upgrade (from versions: none)
     ERROR: No matching distribution found for upgrade

I see that the latest version of ansible-lint is 5.0.6.

I am wondering what the requirements are for upgrading to a newer version of ansible-lint and are those requirements detailed anywhere?
Or am I just not doing the upgrade properly?

I am not a python person and have never used pip prior to installing ansible-lint. So this is most likely a PEBKAC issue and not understanding the bits and pieces involved.

But any assistance would be appreciated. Thank you.

**Comment by ssbarnea** [comment, marked as answer]
You are trying to use a version of python that already reached EOL last year.

Use pip3 instead and read the docs for full command. as you need to specify more than this.

“pip” uses py27 and is also an usupported (boken) version if pip.

**Reply by Errosion** [reply]
Thank you for the quick response!

I figured the python version was most likely the issue. I will be building a python 3.X second installation on this box today and then give this all another go.

You mention needing to specify more than what I had listed for the command. But looking at the documentation, I do not see that. I already have Ansible installed on this system, so I do not need to run the pip commands to install that. Based on the documentation that I can find, which I linked already, it just indicates `pip install "ansible-lint[yamllint]"`. Replacing pip with pip3 in that command once I have the python 3.X version set up.

What more do I need to specify?
Or if you have a link to the documentation, I'll dig through that. I've been using https://ansible-lint.readthedocs.io/en/latest/installing.html
Or are you referring to needing the "[yamllint]" reference in the command?

**Reply by johng42** [reply]
Any updates here?  I am also stuck on ansible-lint 4.2.0 and want to update to 6.0.2 (the  latest at this point) but the docs don't cover the update case.
